### ISSUE TITLE
Add IP route rules to IAC

### ISSUE THREAD

**Issue by ajaxbits** [ajaxbits/config#6]
With my current setup, local traffic will not route properly when `modules.tailscale.enable = true;`.

This is a [known issue](https://tailscale.com/kb/1023/troubleshooting/#linux). 

Currently, I've gotten around the issue by following the above instructions and imperitavely running

```bash
sudo ip rule add to 172.22.0.0/16 priority 2500 lookup main
```

on my node.

I want to add this to IAC somehow. Better yet, parameterize it properly to work with my tailscale module. Best yet, once I've done that, upstream any changes to nixpkgs.

A good solution would:

- Enable an option under `modules.tailscale` that would enable LAN route prioritization
- Either:
  - Dynamically pull in interface names and subnets to handle the rule
  - Add assertions to catch missing interface names and/or subnets at evaluation time.

**Comment by ajaxbits** [comment]
Here are some useful links:

- [My reddit post](https://www.reddit.com/r/Tailscale/comments/16mtrqy/local_network_not_accessible_with/)
- [`routingPolicyRules` lead](https://search.nixos.org/options?channel=23.05&show=systemd.network.networks.<name>.routingPolicyRules&from=0&size=50&sort=relevance&type=packages&query=routingPolicy)
- [A good example of `routingPolicyRules`](https://github.com/KireinaHoro/flakes/blob/5e84785e20e4a4302a8c5a87d4011564c26b0d99/nixos/minato/networking.nix#L126)
- Thought: maybe add nixos options that expose the local network subnet and interface name? How would that work on initial launch?

**Comment by ajaxbits** [comment]
Resolved by changing the routes being advertised by my system. 

Homeassistant used to advertise 172.22.0.0/16. Then, when accept-routes was configured on pat, suddenly it couldn't do it. 

Maybe this is because packets meant for its LAN address were detected to be in the forwarding range, tailscale took over, told the packet to be routed to pat, which then detected the packet to be in the forwarding range, and so on to infinity???

Either way, following the spiritual guidelines laid out by the troubleshooting guide, I turned off subnet routing on Homeassistant and turned on subnet routing for pat, but on the 172.22.0.0/15 cidr to avoid confusion.
